src/io/process.jl

```diff
@@ -75,31 +75,76 @@ Read in the fcs raw, add sample id, subset the columns and transform
 - `fcsRaw`: raw FCS data
 - `md`: Metadata table
 - `panel`: Panel table with a column for Lineage Markers and one for Functional Markers
+- `method`: transformation method, default arcsinh, optional
+- `cofactor`: Cofactor for transformation, default 5, optional
+- `reduce`: Selected only columns which are defined by lineage and functional, optional, 
+    default: true. If false the check for any none columns to be removed (none columns can appear 
+    after concatenating FCS files as well as parameter like: time, event length)
+- `sort`: Sort columns by name to make sure the order when concatinating the dataframes, optional, default: true 
 """
-function createDaFrame(fcsRaw, md, panel)
+function createDaFrame(fcsRaw, md, panel; method = "asinh", cofactor = 5, reduce = true, sort = true)
 
     # extract lineage markers
     lineageMarkers, functionalMarkers = getMarkers(panel)
 
-    transformData(fcsRaw)
+    cc = map(Symbol, vcat(lineageMarkers, functionalMarkers))
+    # markers can be lineage and functional at tthe same time
+    # therefore make cc unique
+    unique!(cc)
 
-    for i in eachindex(md.file_name)
-        df = fcsRaw[md.file_name[i]]
-        insertcols!(df, 1, sample_id = string(md.sample_id[i]))
-    end
+    transformData(fcsRaw, method, cofactor)
 
     dfall = []
-    for (k,v) in fcsRaw
-        push!(dfall,v)
+    colnames = []
+    # for i in eachindex(md.file_name)
+    for (k, v) in fcsRaw
+        # df = fcsRaw[md.file_name[i]]
+        df = v
+        
+        if reduce
+            df = df[:, cc]
+        else
+            # remove the None columns if the columns are not reduced
+            if :None in names(df)
+                delete!(df, :None)
+            end
+            if :None_1 in names(df)
+                delete!(df, :None_1)
+            end
+            if :None_2 in names(df)
+                println(names(df))
+                delete!(df, :None_2)
+            end
+        end
+
+        # sort columns because the order is not garantiert
+        if sort
+            n = names(df)
+            sort!(n)
+            permutecols!(df, n)
+        end
+
+        # insertcols!(df, 1, sample_id = string(md.sample_id[i]))
+        insertcols!(df, 1, sample_id = string(k))
+        push!(dfall,df)
+        # collect the column names of each file for order check
+        push!(colnames, names(df))
+    end
+
+    tmp = colnames[1]
+    for c in colnames
+        if c != tmp
+            println(c)
```

Comment:
Shall we not add a `printLevel` variable, where this verbose output could be muted?